### DISCUSSION TITLE
Search Content by ArcGIS Server URL Dependencies

### DISCUSSION THREAD

**Discussion by character-class** [EsriPS/arcgis-assistant-feedback#62] [Q&A] [answered]
Does ArcGIS Assistant allow searching for content based on _dependencies_?  Given an ArcGIS Server REST URL or, better yet, an ArcGIS Portal Item ID, can admins find maps that are dependent on it?  Then, after finding maps that are dependent on a specific service, can they find dashboards, story maps, web apps, and other Portal items that are dependent on those maps?  Such a feature would allow the admin user to perfectly plan a change in a service.  This is especially helpful for admins who are inheriting a site and don't know how everything is connected.  For example, if a layer will be removed from a service at a future date, the admin can _plan_ by seeing which other items will be affected by the change.  The same is true for the complete removal of a service and helps understand web item connectivity, removal of unused services, and consolidate duplicate services.

For some background, I am the OP for the following question on Esri's Community site:
https://community.esri.com/t5/arcgis-enterprise-ideas/search-content-by-arcgis-server-url/idi-p/1154026

If I have missed something, please let me know, but I could not find a way to perform such a search in Portal or Assistant.  Thank you.

**Comment by slibby** [comment, marked as answer]
HI @character-class I agree this would be great, and incredibly useful for scenarios like you mentioned above. 

Unfortunately there's not a built in ArcGIS REST API that allows you to query this type of question. You'd effectively have to load the item JSON of every single item in the Portal and then scan across it with a full text search. I think there are some glimmers of hope this might come in in the future, but for now there's not a way to do the kind of search you're looking to do.

**Comment by character-class** [comment]
Agreed, but right now this is exactly what I do to reverse engineer our enterprise geodatabase to ArcGIS Server.  I programmatically enter every REST folder (including root) and load the JSON manifest of every service.  I inspect it to find which services use a particular feature class or table or even a sibling of it, and return that list from the function.  While it takes a while to run, it is more effective than manually opening each document that represents the service, viewing each layer's properties, and documenting the results.

The purpose of the script, of course, is to see which services would be affected by a change to the schema of a feature class or table, most notably, the deletion of a field, "renaming" a field, or changing the values of a coded domain field.

Here is the script I posted on GeoNet which does the above.  The latest script is towards the end of the thread.
https://community.esri.com/t5/publishing-and-managing-services-questions/listing-feature-classes-used-by-arcgis-server

The script does not, however, utilize advanced concepts like finding _views_ of those feature classes and table, or finding SQL query layers either, but it works for my purposes, so I shared it.  My posts there have received seven kudos.

So, could anything like that be run for Portal?  I'd love to get a list of maps that depend on a web item, and a list of apps, dashboards, story boards, etc. that depend on those maps.  It would be great for planning purposes, even if it used a brute force method to determine those connections.
